lib/tools/system-commands.js

```diff
@@ -107,10 +125,52 @@ systemCallMethods.connect = async function connect () {
   }
   log.info(`Successfully established the connection to ${IDB_EXECUTABLE} service for '${this.udid}'`);
 
+  if (onlineTimeout) {
+    await this.waitForDevice(onlineTimeout);
+  }
+
   this.executable.path = binaryPaths[IDB_EXECUTABLE];
   this.companion.path = binaryPaths[IDB_COMPANION_EXECUTABLE];
 };
 
+/**
+ * Blocks until the device under test starts responding to idb commands.
+ * The device must be booted/online and idb must be already connected for that to happen
+ *
+ * @param {?number} timeoutMs [10000] - The number of milliseconds to wait
+ * until the device under tests is online. The method will return immediately
+ * if the timeout is falsy
+ * @throws {Error} if the device is not responding within the given timeout
+ */
+systemCallMethods.waitForDevice = async function waitForDevice (timeoutMs = 10000) {
+  if (!timeoutMs) {
+    log.debug('No timeout is provided, so not waiting until the device is online');
+    return;
+  }
+
+  log.debug(`Waiting ${timeoutMs}ms until the device is online`);
+  const started = process.hrtime();
+  try {
+    await waitForCondition(async () => {
+      try {
+        await tpExec(IDB_EXECUTABLE, ['describe', '--json', '--udid', this.udid]);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    }, {
+      waitMs: timeoutMs,
+      intervalMs: 300,
+    });
+  } catch (e) {
+    throw new Error(`The device '${this.udid}' is not responding to idb requests after ${timeoutMs}ms timeout. ` +
+      `Original error: ${e.stderr || e.message}`);
```

Comment:
Put `'` surround the original error message?

Reply:
the colon is enough to separate it from the static context